pennylane_lightning/lightning_qubit.py

```diff
@@ -49,9 +49,7 @@ class LightningQubit(DefaultQubit):
     version = __version__
     author = "Xanadu Inc."
 
-    operations = {
-        "BasisState",
-        "QubitStateVector",
+    kernel_operations = {
```

Comment:
By doing this, `lightning.qubit` will now support any gate that `default.qubit` does. Further, if the gate is in `kernel_operation` then we have an optimized kernel for it. If not, we'll directly use the the gate's matrix to apply the gate through standard matrix multiplication.